packages/navigation/src/withNavigation.js

```diff
@@ -8,14 +8,16 @@ import type { Navigation, Route } from './types';
 const navigate = (url: Route, params?: Object) =>
   Router.push({ pathname: `/${url}`, query: params });
 
-type Props = {};
-type EnhancedProps = { +navigation: Navigation };
+type Props = {
+  navigation: Navigation | void,
+};
 
-export const withNavigation = (
-  Component: React.ComponentType<EnhancedProps>,
-): React.ComponentType<Props> =>
-  class extends React.Component<Props> {
+export const withNavigation = <PassedProps: {}>(
+  WrappedComponent: React.ComponentType<PassedProps>,
```

Comment:
Thanks for improving this, that seems more like `flow`'s way of doing it 👍

Reply:
👍